src/analysis/convex/specific_functions.lean

```diff
@@ -114,3 +114,43 @@ begin
     apply mul_nonneg (le_trans zero_le_one hp),
     exact mul_nonneg (sub_nonneg_of_le hp) (rpow_nonneg_of_nonneg (le_of_lt hx) _) }
 end
+
+lemma concave_on_log_Ioi : concave_on (Ioi 0) log :=
+begin
+  have h₁ : Ioi 0 ⊆ ({0} : set ℝ)ᶜ,
+  { intros x hx hx',
+    rw [mem_singleton_iff] at hx',
+    rw [hx'] at hx,
+    exact lt_irrefl 0 hx },
+  refine concave_on_open_of_deriv2_nonpos (convex_Ioi 0) is_open_Ioi _ _ _,
+  { exact differentiable_on_log.mono h₁ },
+  { refine differentiable_on.mono _ h₁,
+    rw [deriv_log'],
+    refine differentiable_on_id.inv (λ x hx, _),
+    rwa [compl_singleton_eq (0 : ℝ)] at hx },
+  { intros x hx,
+    rw [function.iterate_succ, function.iterate_one],
+    change (deriv (deriv log)) x ≤ 0,
+    rw [deriv_log', deriv_inv (show x ≠ 0, by {rintro rfl, exact lt_irrefl 0 hx})],
+    exact neg_nonpos.mpr (inv_nonneg.mpr (pow_two_nonneg x)) }
```

Comment:
I just wrote a quick PR #5521, which will let this be simplified to 
```
  { intros x hx,
    simp [function.iterate_succ, function.iterate_one, pow_two_nonneg] }
```
Let's not hold your PR up for this though.

Reply:
Nice!